src/main/java/tigase/xmpp/XMPPIOService.java

```diff
@@ -328,12 +331,24 @@ public long getTotalPacketsSent() {
 		return totalPacketsSent;
 	}
 
+	public Optional<JID> getAuthorisedUserJid() {
+		return Optional.ofNullable(authorisedUserJid);
+	}
+
+	public void setAuthorisedUserJid(JID authorisedUserJid) {
+		this.authorisedUserJid = authorisedUserJid;
+	}
+
+	@Deprecated
+	@TigaseDeprecated(removeIn = "9.0.0", since = "8.2.0", note = "#getAuthorisedUserJid should be used instead")
 	public String getUserJid() {
-		return this.jid;
+		return getAuthorisedUserJid().isPresent() ? getAuthorisedUserJid().get().toString() : null;
```

Comment:
Why not use Optional.map().orElse(null)?

Also, I recall that this string was BareJID and now it will be full JID. Please verify what result was (bare jid or a full jid) and check impact of this change on the server. If that was bare JID I would suggest calling `getBareJID()` before calling `toString()` to maintain API stability.

Reply:
As explained in different places the contents is variable (BareJID or FullJID) and as string it's then later on parsed thus it doesn't make any difference here.